caas/kubernetes/provider/k8s.go

```diff
@@ -1640,6 +1666,41 @@ func (k *kubernetesClient) WatchUnits(appName string) (watcher.NotifyWatcher, er
 	return k.newWatcher(w, appName, k.clock)
 }
 
+// WatchService returns a watcher which notifies when there
+// are changes to the deployment of the specified application.
+func (k *kubernetesClient) WatchService(appName string) (watcher.NotifyWatcher, error) {
+	// Application may be a statefulset or deployment. It may not have
+	// been set up when the watcher is started so we don't know which it
+	// is ahead of time. So use a multi-watcher to cover both cases.
+	statefulsets := k.AppsV1().StatefulSets(k.namespace)
+	sswatcher, err := statefulsets.Watch(v1.ListOptions{
+		LabelSelector: applicationSelector(appName),
```

Comment:
probably we should include `IncludeUninitialized=true`?

Reply:
Not sure, we don't elsewhere for watching pods. If we do decide to change, should do so everyhwere else too.